### PR TITLE
chore: introduce detailed breaking sections in CHANGELOG

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

This patch would introduce following sections into CHANGELOG.md.
- Event Breaking Change
- State Machine Breaking

## Motivation and context

Our client considers the changes in the events as Client Breaking. Hence, we added another section, "Event Breaking" and will keep tracking on it.

## Checklist:

- [ ] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by jaeseung-bae — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,20 +55,22 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
 ### Breaking Changes
 * (x/foundation) [\#999](https://github.com/Finschia/finschia-sdk/pull/999) migrate x/foundation FoundationTax into x/params
-* (refactor) [\#1090](https://github.com/Finschia/finschia-sdk/pull/1090) Automate EventTypeMessage inclusion in every message execution (backport #1063)
-* (x/bank) [#1093](https://github.com/Finschia/finschia-sdk/pull/1093) Remove message events including `sender` attribute whose information is already present in the relevant events (backport #1066)
 * (ostracon) [\#1099](https://github.com/Finschia/finschia-sdk/pull/1099) feat!: remove libsodium vrf library.
 * (x/collection) [\#1102](https://github.com/finschia/finschia-sdk/pull/1102) Reject modifying NFT class with token index filled in MsgModify
 * (x/collection) [\#1105](https://github.com/Finschia/finschia-sdk/pull/1105) Add minted coins to balance in x/collection MsgMintFT
 * (x/collection) [\#1106](https://github.com/Finschia/finschia-sdk/pull/1106) Support x/collection migration on old chains
 
+### Event Breaking
```
Comment (0Tech) [inline CHANGELOG.md:69]:
```suggestion
### Event Breaking Changes
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -54,20 +55,22 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
 ### Removed
 
 ### Breaking Changes
 * (x/foundation) [\#999](https://github.com/Finschia/finschia-sdk/pull/999) migrate x/foundation FoundationTax into x/params
-* (refactor) [\#1090](https://github.com/Finschia/finschia-sdk/pull/1090) Automate EventTypeMessage inclusion in every message execution (backport #1063)
-* (x/bank) [#1093](https://github.com/Finschia/finschia-sdk/pull/1093) Remove message events including `sender` attribute whose information is already present in the relevant events (backport #1066)
 * (ostracon) [\#1099](https://github.com/Finschia/finschia-sdk/pull/1099) feat!: remove libsodium vrf library.
 * (x/collection) [\#1102](https://github.com/finschia/finschia-sdk/pull/1102) Reject modifying NFT class with token index filled in MsgModify
 * (x/collection) [\#1105](https://github.com/Finschia/finschia-sdk/pull/1105) Add minted coins to balance in x/collection MsgMintFT
 * (x/collection) [\#1106](https://github.com/Finschia/finschia-sdk/pull/1106) Support x/collection migration on old chains
```
Comment (zemyblue) [inline CHANGELOG.md:67]: Please classify these parts appropriately as well.

Comment (0Tech) [reply CHANGELOG.md]: I added `State Machine Breaking` section into the changelog. I followed the naming in the upstream's. That's the reason why the name of the section is not `State Machine Breaking Changes`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -54,20 +55,24 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
-* (baseapp) [\#1092](https://github.com/finschia/finschia-sdk/pull/1092) Do not add `module` attribute in case of ibc messages (backport #1079)
 * (x/foundation) [\#1108](https://github.com/Finschia/finschia-sdk/pull/1108) Rollback MsgUpdateParams parts from #999
 
 ### Removed
 
 ### Breaking Changes
-* (x/foundation) [\#999](https://github.com/Finschia/finschia-sdk/pull/999) migrate x/foundation FoundationTax into x/params
-* (refactor) [\#1090](https://github.com/Finschia/finschia-sdk/pull/1090) Automate EventTypeMessage inclusion in every message execution (backport #1063)
-* (x/bank) [#1093](https://github.com/Finschia/finschia-sdk/pull/1093) Remove message events including `sender` attribute whose information is already present in the relevant events (backport #1066)
 * (ostracon) [\#1099](https://github.com/Finschia/finschia-sdk/pull/1099) feat!: remove libsodium vrf library.
```
Comment (zemyblue) [inline CHANGELOG.md:63]: How about moving to `Improvements`, it just modify build and ci option because libsodium does not use it.

**Review by zemyblue — APPROVED** [review]

**Review by jaeseung-bae — APPROVED** [review]

**Pull request merged**